Add a governance extension that implements super quorum (#5492)
OpenZeppelin/openzeppelin-contracts: contracts/governance/README.adoc

* {GovernorVotesQuorumFraction}: Combines with `GovernorVotes` to set the quorum as a fraction of the total token supply.

> * {GovernorVotesSuperQuorumFraction}: Combines with `GovernorVotesQuorumFraction` to set the super quorum as a fraction of the total token supply.

Reviewer: ```suggestion
* {GovernorVotesSuperQuorumFraction}: Combines `GovernorSuperQuorum` with `GovernorVotesQuorumFraction` to set the super quorum as a fraction of the total token supply.
```

* {GovernorProposalGuardian}: Adds a proposal guardian that can cancel proposals at any stage in their lifecycle--this permission is passed on to the proposers if the guardian is not set.

> * {GovernorSuperQuorum}: Adds a super quorum that if reached, the proposal can proceed to the next state without waiting for the proposal deadline.
> 

Reviewer 2: ```suggestion
* {GovernorSuperQuorum}: Extension of {Governor} with a super quorum. Proposals that meet the super quorum (and have a majority of for 
 * votes) advance to the `Succeeded` state before the proposal deadline.

```

* {GovernorProposalGuardian}: Adds a proposal guardian that can cancel proposals at any stage in their lifecycle--this permission is passed on to the proposers if the guardian is not set.

> * {GovernorSuperQuorum}: Extension of {Governor} with a super quorum. Proposals that meet the super quorum (and have a majority of for 
>  * votes) advance to the `Succeeded` state before the proposal deadline.

Reviewer 2: ```suggestion
* {GovernorSuperQuorum}: Extension of {Governor} with a super quorum. Proposals that meet the super quorum (and have a majority of for votes) advance to the `Succeeded` state before the proposal deadline.
```
